lib/vrl/web-playground/public/vrl-highlighter.js

```diff
@@ -0,0 +1,636 @@
+                    "@default": { token: "string", next: "@string.'" },
+                    },
+                },
+            ],
+        ],
+
+        string: [
+            [/[^"'\#\\]+/, "string"],
+            [/@escapes/, "string.escape"],
+            [/\./, "string.escape.invalid"],
+            [/\./, "string.escape.invalid"],
+
+            [
+                /#{/,
+                {
+                    cases: {
+                    '$S2=="': {
+                        token: "string",
+                        next: "root.interpolatedstring",
+                    },
+                    "@default": "string",
+                    },
+                },
+            ],
+
+            [
+                /["']/,
+                {
+                    cases: {
+                    "$#==$S2": { token: "string", next: "@pop" },
+                    "@default": "string",
+                    },
+                },
+            ],
+            [/#/, "string"],
+        ],
+
+        herestring: [
+            [
+                /("""|''')/,
+                {
+                    cases: {
+                    "$1==$S2": { token: "string", next: "@pop" },
+                    "@default": "string",
+                    },
+                },
+            ],
+            [/[^#\\'"]+/, "string"],
+            [/['"]+/, "string"],
+            [/@escapes/, "string.escape"],
+            [/\./, "string.escape.invalid"],
+
+            [/#{/, { token: "string.quote", next: "root.interpolatedstring" }],
+            [/#/, "string"],
+        ],
+
+        comment: [
+            [/[^#]+/, "comment"],
+            [/###/, "comment", "@pop"],
+            [/#/, "comment"],
+        ],
+
+        hereregexp: [
+            [/[^\\\/#]+/, "regexp"],
+            [/\\./, "regexp"],
+            [/#.*$/, "comment"],
+            ["///[igm]*", { token: "regexp", next: "@pop" }],
+            [/\//, "regexp"],
+        ],
+
+        function_arg: [
+            [
+                /([a-zA-Z_!]+)\(/,
+                { cases: { $S1: { token: "keyword", next: "@push", log: "function_arg_nested::\n\n$0" } } },
+            ],
+            // we copy the string highlighters from root and apply them to function arguments as well
+            [/"""/, "string", '@herestring."""'],
+            [/'''/, "string", "@herestring.'''"],
+            [
+                /"/,
+                {
+                    cases: {
+                    "@eos": "string",
+                    "@default": { token: "string", next: '@string."' },
+                    },
+                },
+            ],
+            [
+                /'/,
+                {
+                    cases: {
+                    "@eos": "string",
+                    "@default": { token: "string", next: "@string.'" },
+                    },
+                },
+            ],
+            // TODO: field access inside function parameters sometimes fails, eg: in a
+            // newline func(.foo)
```

Comment:
It might be related, but here's one I stumbled across that shows different syntax highlighting with/without a space at the end:

https://deploy-preview-15557--vrl-playground.netlify.app/?state=eyJwcm9ncmFtIjoicGFyc2VfaW50ISguaHR0cF9zdGF0dXMpXG5wYXJzZV9pbnQhKC5odHRwX3N0YXR1cykgIiwiZXZlbnQiOnsibWVzc2FnZSI6IkhlbGxvIFZSTCIsImZvbyI6ImRlbGV0ZSBtZSIsImh0dHBfc3RhdHVzIjoiMjAwIn0sImlzX2pzb25sIjpmYWxzZX0%3D

Also note that in the last example `)` is shown as part of the path syntax, which is only valid if there's also an opening `(` in the path.